### PR TITLE
Custom status expiry

### PR DESCRIPTION
**Pull request by manojmalik20**
#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

File: app/components/custom_status/custom_status_emoji.tsx [hunk 1]
```diff
@@ -29,15 +28,17 @@ const CustomStatusEmoji = ({emojiSize, userID, style, testID}: ComponentProps) =
...
+        customStatus && (
+            <Text
+                style={style}
+                testID={`${testIdPrefix}custom_status_emoji.${customStatus.emoji}`}
+            >
+                <Emoji
+                    size={emojiSize}
+                    emojiName={customStatus.emoji}
+                />
+            </Text>
```
Comment (hrishavkmr) [inline app/components/custom_status/custom_status_emoji.tsx]: This render logic should be outside the `return()` in a variable.
```code
const customEmoji = customStatus ? <Text>...</Text> : null;
return (
     {customEmoji}
)
```

File: app/components/custom_status/custom_status_expiry.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    const showDay = !useTime && useDay && (
+        <FormattedDate
+            format='dddd'
+            timezone={timezone}
+            value={expiryMomentTime}
+            style={styleProp || styles.text}
+        />
+    );
+
+    const showDate = !isToday && !useTime && !useDay && (
```
Comment (hrishavkmr) [inline app/components/custom_status/custom_status_expiry.tsx]: condition can be simplified
```suggestion
    const showDate = !(isToday || useTime || useDay) && (
```

File: app/screens/custom_status_clear_after/date_time_selector.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            {show && (
+                <DateTimePicker
+                    testID='dateTimePicker'
+                    value={moment(date).toDate()}
+                    mode={mode}
+                    is24Hour={militaryTime}
+                    display={Platform.OS === 'ios' ? 'spinner' : 'default'}
+                    onChange={onChange}
+                />
+            )}
```
Comment (hrishavkmr) [inline app/screens/custom_status_clear_after/date_time_selector.tsx]: store this outside the render function in a variable and render that variable here.

File: app/screens/custom_status/custom_status_suggestion.tsx [hunk 4]
```diff
@@ -56,11 +61,22 @@ const CustomStatusSuggestion = (props: Props) => {
...
+                                theme={theme}
+                                textStyle={{color: theme.centerChannelColor}}
+                            />
+                        </View>
+                        <View style={style.expiryTimeContainer}>
+                            {duration && (
+                                <CustomStatusText
+                                    text={intl.formatMessage(durationValues[duration])}
+                                    theme={theme}
+                                    textStyle={{color: changeOpacity(theme.centerChannelColor, 0.6)}}
```
Comment (hrishavkmr) [inline app/screens/custom_status/custom_status_suggestion.tsx]: create a new style in the stylesheet and use that style here.

File: app/screens/channel_info/index.js [hunk 5]
```diff
@@ -45,13 +49,15 @@ function makeMapStateToProps() {
             }
             customStatusEnabled = isCustomStatusEnabled(state);
             customStatus = customStatusEnabled ? getCustomStatus(state, teammateId) : undefined;
```
Comment (chetanyakan) [inline app/screens/channel_info/index.js]:
```suggestion
            customStatus = customStatusEnabled && getCustomStatus(state, teammateId);
```

File: app/components/custom_status/custom_status_expiry.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+import {useSelector} from 'react-redux';
+import {Text, TextStyle} from 'react-native';
+import FormattedText from '@components/formatted_text';
+import {Theme} from '@mm-redux/types/preferences';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {getCurrentUserTimezone} from '@mm-redux/selectors/entities/timezone';
+type Props = {
+    theme: Theme;
+    time: Date;
+    styleProp?: TextStyle;
```
Comment (chetanyakan) [inline app/components/custom_status/custom_status_expiry.tsx]: Rename this prop to `textStyles`

File: app/components/custom_status/custom_status_expiry.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+                id='custom_status.expiry.until'
+                defaultMessage='Until'
+            />{' '}
+        </Text>
+    ) : null;
+
+    return (
+        <Text
+            testID={props.testID}
+            style={styleProp || styles.text}
```
Comment (chetanyakan) [inline app/components/custom_status/custom_status_expiry.tsx]: I wonder if we should use something like this:

```suggestion
            style={{...styles.text, ...textStyles}}
```

File: app/components/custom_status/custom_status_expiry.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        >
+            {withinBrackets ? '(' : null}
```
Comment (chetanyakan) [inline app/components/custom_status/custom_status_expiry.tsx]: nit -  I wonder if using `&&` here makes more sense.

```suggestion
            {withinBrackets && '('}
```

File: app/screens/channel_info/channel_info_header.js [hunk 9]
```diff
@@ -157,6 +161,18 @@ export default class ChannelInfoHeader extends React.PureComponent {
...
+            (
+                <Text style={style.customStatusExpiry}>
+                    <CustomStatusExpiry
+                        time={customStatus.expires_at}
+                        theme={theme}
+                        styleProp={style.customStatusExpiry}
+                        showPrefix={true}
+                    />
+                </Text>
+            ) : null;
```
Comment (chetanyakan) [inline app/screens/channel_info/channel_info_header.js]: Should we add a check here for `!isCustomStatusExpired`?

**Review by chetanyakan — APPROVED** [review]

**Comment by manojmalik20** [comment]
Corresponding PR has been made in mattermost.

**Pull request closed without merging**